spec/rubocop/yaml_duplication_checker_spec.rb

```diff
@@ -33,8 +33,10 @@ def check(yaml, &block)
         key1 = key_a
         key2 = key_b
       end
-      expect(key1.start_line).to eq(0)
-      expect(key2.start_line).to eq(3)
+      if key1.respond_to?(:start_line)
```

Comment:
We also have mechanism to run different specs on different Ruby versions. I think it'd be better to use it.

Reply:
can you point me to a spec which uses this? then I can adapt the PR

Reply:
See `erb_new_arguments_spec.rb`.

Reply:
hmm do I need to do anything more then set the rspec tags to make this work? it still seems to run all tests regardless which ruby version i'm using and therefore I get failing tests on old ruby version and errors on newer ones because of the changed output

Reply:
That's weird. I get the same results. I wonder if the RSpec functionality for specifying which Ruby version a certain scope is valid for has changed. This works:
```diff
diff --git a/spec/rubocop/yaml_duplication_checker_spec.rb b/spec/rubocop/yaml_duplication_checker_spec.rb
index 068270128..b6dc8548f 100644
--- a/spec/rubocop/yaml_duplication_checker_spec.rb
+++ b/spec/rubocop/yaml_duplication_checker_spec.rb
@@ -26,7 +26,7 @@ RSpec.describe RuboCop::YAMLDuplicationChecker do
 
     include_examples 'call block'
 
-    context '< Ruby 2.5', :ruby24 do
+    context '< Ruby 2.5', if: RUBY_VERSION >= '2.4' do
       it 'calls block with keys' do
         key1 = nil
         key2 = nil
@@ -39,7 +39,7 @@ RSpec.describe RuboCop::YAMLDuplicationChecker do
       end
     end
 
-    context '>= Ruby 2.5', :ruby25 do
+    context '>= Ruby 2.5', if: RUBY_VERSION >= '2.5' do
       it 'calls block with keys' do
         key1 = nil
         key2 = nil
```
so maybe we should change `:ruby2x` everywhere?

Reply:
ok I'll change my tests then to this syntax

Reply:
> That's weird. I get the same results. I wonder if the RSpec functionality for specifying which Ruby version a certain scope is valid for has changed. This works:

@Darhazer @bquorning ^^

Reply:
fixed for the tests I added/touched. For the other cases it doesn't seem to be a problem

Reply:
Setting the ruby version via `:ruby25` actually just sets the `ruby_version` variable, which is used by the ProcessedSource, e.g. what parser version to be used. It's useful for cops specs, but here you are not testing a cop.